Snowplow/SPSQLiteEventStore.m

```diff
@@ -133,53 +129,79 @@ - (NSUInteger) count {
     return num;
 }
 
-- (NSDictionary *) getEventWithId:(long long int)id_ {
-    __block NSDictionary *dict = nil;
-    [_queue inDatabase:^(FMDatabase *db) {
+- (NSArray<SPEmitterEvent *> *)emittableEventsWithQueryLimit:(NSUInteger)queryLimit {
+    return [self getAllEventsLimited:self.sendLimit];
+}
+
+// MARK: SPSQLiteEventStore methods
+
+- (BOOL) createTable {
+    __block BOOL res = NO;
+    [self.queue inDatabase:^(FMDatabase *db) {
+        if ([db open]) {
+            res = [db executeStatements:_queryCreateTable];
+        }
+    }];
+    return res;
+}
+
+- (long long int) insertEvent:(SPPayload *)payload {
+    return [self insertDictionaryData:[payload getAsDictionary]];
+}
+
+- (long long int) insertDictionaryData:(NSDictionary *)dict {
+    __block long long int res = -1;
+    if (!dict) {
+      return res;
+    }
+    [self.queue inDatabase:^(FMDatabase *db) {
+        if ([db open]) {
+            NSData *data = [NSJSONSerialization dataWithJSONObject:dict options:0 error:nil];
+            [db executeUpdate:_queryInsertEvent, data];
+            res = (long long int) [db lastInsertRowId];
+        }
+    }];
+    return res;
+}
+
+- (SPEmitterEvent *) getEventWithId:(long long int)id_ {
+    __block SPEmitterEvent *event = nil;
+    [self.queue inDatabase:^(FMDatabase *db) {
         if ([db open]) {
             FMResultSet *s = [db executeQuery:_querySelectId, [NSNumber numberWithLongLong:id_]];
             while ([s next]) {
                 NSData * data = [s dataForColumn:@"eventData"];
-                SPLogDebug(@"Item: %@ %@ %@",
-                     [NSNumber numberWithInt:[s intForColumn:@"ID"]],
-                     [s dateForColumn:@"dateCreated"],
-                     [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding]);
-                dict = [NSJSONSerialization JSONObjectWithData:data options:0 error:0];
+                NSDictionary *dict = [NSJSONSerialization JSONObjectWithData:data options:0 error:0];
+                SPPayload *payload = [[SPPayload alloc] initWithNSDictionary:dict];
+                event = [[SPEmitterEvent alloc] initWithPayload:payload storeId:id_];
```

Comment:
You're losing precision here. SPEmitter event should accept a `long long` for the `storeId`.
You can also use `long long` rather than `long long int`, same thing and saves typing 4 characters 😉

Reply:
damned archaic languages!